src/overlays/actors/ovl_Bg_Treemouth/z_bg_treemouth.c

```diff
@@ -21,29 +37,215 @@ const ActorInit Bg_Treemouth_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/BgTreemouth_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/func_808BC65C.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/func_808BC6F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/func_808BC80C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/func_808BC864.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/func_808BC8B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/func_808BC9EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/func_808BCAF0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/func_808BCB8C.s")
+static InitChainEntry initChain[] = {
+    ICHAIN_U8(unk_1F, 5, ICHAIN_CONTINUE),     ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F4, 8000, ICHAIN_CONTINUE), ICHAIN_F32(unk_F8, 300, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 300, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/BgTreemouth_Update.s")
+// unused
+u32 D_808BD9C4[] = {
+    0xC52BA000, 0x44084000, 0x4592B000, 0xC525E000, 0x43120000, 0x458DB000,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Treemouth/BgTreemouth_Draw.s")
+extern u32 D_060009D0;
+extern u32 D_06000E94;
+
+void BgTreemouth_SetupAction(BgTreemouth* this, BgTreemouthActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgTreemouth_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgTreemouth* this = THIS;
+    u32 localC = 0;
+
+    Actor_ProcessInitChain(thisx, initChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_06000E94, &localC);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, localC);
+    ActorShape_Init(&thisx->shape, 0.0f, NULL, 0.0f);
+    Actor_SetHeight(thisx, 50.0f);
+
+    if ((gSaveContext.sceneSetupIndex < 4) && (LINK_IS_CHILD)) {
+        BgTreemouth_SetupAction(this, func_808BC8B8);
+    } else {
+        if ((LINK_IS_ADULT) || (gSaveContext.sceneSetupIndex == 7)) {
+            this->unk_168 = 0.0f;
+            BgTreemouth_SetupAction(this, func_808BCB8C);
+        } else {
+            this->unk_168 = 1.0f;
+            BgTreemouth_SetupAction(this, func_808BC6F8);
+        }
+    }
+
+    thisx->textId = 0x905;
+}
+
+void BgTreemouth_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgTreemouth* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void func_808BC65C(BgTreemouth* this, GlobalContext* globalCtx) {
+    CsCmdActorAction* actorAction;
+
+    if ((globalCtx->csCtx.state != 0)) {
+        actorAction = globalCtx->csCtx.npcActions[0];
+        if (actorAction != NULL) {
```

Comment:
actorActions was recently renamed npcActions so I would change these throughout the file